yt_dlp/extractor/biqle.py

```diff
@@ -32,54 +34,79 @@ class BIQLEIE(InfoExtractor):
+            webpage, 'global Embed url', flags=re.DOTALL)
+        hash = self._search_regex(
+            r'<script id="data-embed-video.+?hash: "([^"]+)"[^<]*</script>',
+            webpage, 'Hash', flags=re.DOTALL)
+
+        embed_url = globalEmbed_url + hash
+        self.write_debug('embed_url: %s' % embed_url)
+
         if VKIE.suitable(embed_url):
             return self.url_result(embed_url, VKIE.ie_key(), video_id)
 
         embed_page = self._download_webpage(
-            embed_url, video_id, headers={'Referer': url})
-        video_ext = self._get_cookies(embed_url).get('video_ext')
-        if video_ext:
-            video_ext = compat_urllib_parse_unquote(video_ext.value)
-        if not video_ext:
-            video_ext = compat_b64decode(self._search_regex(
-                r'video_ext\s*:\s*[\'"]([A-Za-z0-9+/=]+)',
-                embed_page, 'video_ext')).decode()
-        video_id, sig, _, access_token = video_ext.split(':')
+            embed_url, video_id, 'Downloading embed webpage', headers={'Referer': url})
+
+        globParams = self._parse_json(self._search_regex(
+            r'<script id="globParams">.*window.globParams = ([^;]+);[^<]+</script>',
+            embed_page, 'Global Parameters', flags=re.DOTALL), video_id, transform_source=js_to_json)
+        self.write_debug('globParams: %s' % globParams)
+        hostName = compat_b64decode(globParams['server'][::-1]).decode()
+        server = 'https://%s/method/video.get/' % hostName
+        self.write_debug('server: %s' % server)
+
         item = self._download_json(
-            'https://api.vk.com/method/video.get', video_id,
-            headers={'User-Agent': 'okhttp/3.4.1'}, query={
-                'access_token': access_token,
-                'sig': sig,
-                'v': 5.44,
+            server + video_id, video_id,
+            headers={'Referer': url}, query={
+                'token': globParams['video']['access_token'],
                 'videos': video_id,
+                'ckey': globParams['c_key'],
+                'credentials': globParams['video']['credentials'],
             })['response']['items'][0]
-        title = item['title']
 
         formats = []
         for f_id, f_url in item.get('files', {}).items():
             if f_id == 'external':
                 return self.url_result(f_url)
             ext, height = f_id.split('_')
-            formats.append({
-                'format_id': height + 'p',
-                'url': f_url,
-                'height': int_or_none(height),
-                'ext': ext,
-            })
+            if globParams['video']['partial']['quality'].get(height) is not None:
+                formats.append({
+                    'format_id': height + 'p',
+                    'url': f_url.replace('https://', 'https://%s/' % hostName) + '&videos=%s' % video_id + '&extra_key=%s' % globParams['video']['partial']['quality'][height],
```

Comment:
use `%`/`format`/`fstrings`, not `+`

Reply:
Sry, but i dont understand how can I do that. (first few weeks of coding in python)
I need to place a new url right after https://, and need to concatenate more strings at the end.
How should i do that without +?

I can think concatenating first then replacing, but for me the code is so obscure that dont makes sense.
`('%s&videos=%s&extra_key=%s' % (f_url,video_id, extra_key)).replace('https://', 'https://%s/' % hostName)`

Reply:
Even the JS of the page assumes that, so I think we are safe.

![image](https://user-images.githubusercontent.com/216170/153734787-6f9dd865-6199-4763-93c3-ba69a6fe9038.png)